crates/iota-grpc-api/src/bcs_event.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! BCS serialization utilities for IotaEvent.
+//!
+//! IotaEvent cannot be directly BCS serialized because it uses
+//! serde_json::Value, and annotations like #[serde(flatten)].
+
+use iota_json_rpc_types::IotaEvent;
+use iota_types::{
+    base_types::{IotaAddress, ObjectID},
+    event::EventID,
+};
+use move_core_types::{identifier::Identifier, language_storage::StructTag};
+use serde::{Deserialize, Serialize};
+
+/// Convert IotaEvent to BCS bytes
+pub fn try_to_bcs_bytes(event: &IotaEvent) -> Result<Vec<u8>, bcs::Error> {
+    let bcs_event = BcsIotaEvent::from(event.clone());
+    bcs::to_bytes(&bcs_event)
```

Comment:
```suggestion
    bcs::to_bytes(&bcs_event).unwrap()
```
It seems that `bcs::to_bytes` fails in really exceptional cases.

Reply:
Thanks for pointing this out. After refactory I removed the whole bcs_event module in 8dc4a7a61d433c863eaa023bdf32d1329fd25262